packages/org.json_schema.contrib/internal/TypesGenerator.pkl

```diff
@@ -669,5 +669,100 @@ local function generateBaseType(schema: JsonSchema, typeNames: Type.TypeNames):
+          else elem.`const` ?? res.`const`
+
+          // number
+          multipleOf = if (elem.multipleOf != null && res.multipleOf != null)
+            if (elem.multipleOf is Int && res.multipleOf is Int) math.lcm(elem.multipleOf!! as Int, res.multipleOf!! as Int)
+            else allOfErr("multiple non-integer multipleOf")
+          else elem.multipleOf ?? res.multipleOf
+          minimum = collateMin(elem.minimum, res.minimum)
+          exclusiveMinimum = collateMin(elem.exclusiveMinimum, res.exclusiveMinimum)
+          maximum = collateMax(elem.maximum, res.maximum)
+          exclusiveMaximum = collateMax(elem.exclusiveMaximum, res.exclusiveMaximum)
+
+          // string
+          pattern = if (elem.pattern != null && res.pattern != null) "\(elem.pattern)|\(res.pattern)"
+            else elem.pattern ?? res.pattern
+          minLength = collateMin(elem.minLength, res.minLength) as UInt?
+          maxLength = collateMax(elem.maxLength, res.maxLength) as UInt?
+          format = if (res.format != null && elem.format != null && res.format != elem.format) allOfErr("conflicting format")
+          else elem.format ?? res.format
+
+          // object
+          properties = collateProperties(elem.properties, res.properties)
+          patternProperties = collateProperties(elem.patternProperties, res.patternProperties)
+          additionalProperties = collateSchema(elem.additionalProperties, res.additionalProperties, "additionalProperties")
+          required = if (elem.required == null && res.required == null) null
+          else ((elem.required?.toSet() ?? Set()) + (res.required?.toSet() ?? Set())).toListing()
+          propertyNames = collateSchema(elem.propertyNames, res.propertyNames, "propertyNames")
+          minProperties = collateMin(elem.minProperties, res.minProperties) as UInt?
+          maxProperties = collateMax(elem.maxProperties, res.maxProperties) as UInt?
+
+          // array
+          items = if (elem.items == null && res.items == null) null
+          else if ((res.items == null) != (elem.items == null)) elem.items ?? res.items
+          else if (res.items is JsonSchema && elem.items is JsonSchema) collateAllOf(new JsonSchema { allOf { res.items; elem.items as JsonSchema } })
+          else allOfErr("conflicting items lhs:'\(elem.items)' rhs:'\(res.items)'")
+          additionalItems = collateSchema(elem.additionalItems, res.additionalItems, "additionalItems")
+          minItems = collateMin(elem.minItems, res.minItems) as UInt?
+          maxItems = collateMax(elem.maxItems, res.maxItems) as UInt?
+          uniqueItems = if (elem.uniqueItems == true || res.uniqueItems == true) true
+            else elem.uniqueItems ?? res.uniqueItems
+
+          // composition
+          oneOf = collateSet(elem.oneOf, res.oneOf)
+          anyOf = collateSet(elem.anyOf, res.anyOf)
+          not = collateSchema(elem.not, res.not, "not")
```

Comment:
There's some things that are missing that would be good to collate:

* title
* description
* default
* deprecated
* examples
* readOnly
* writeOnly

Reply:
How would each of these be collated? I didn't address these because I wasn't sure there was a right way. Here's how I might do these:
* title - join by newlines? commas?
* description - join by newlines
* default - error if more than one value set
* deprecated - I don't think there's a valid way to do this
* examples - if all elements with non-null examples are listings, append them, otherwise error
* readOnly - all sub-schemas must set the same readOnly value? all sub-schemas with non-null readOnly must set the same value?
* writeOnly - same as readOnly

Reply:
My suggestions (I'm making this up too, so feel free to push back)

* title, description: maybe join by `\n---\n`? That would create a line break in the markdown doc comment.
* default: makes sense
* examples: concat into a listing
* readOnly, writeOnly: your suggestion makes sense
* deprecated: same treatment as readOnly, writeOnly

Reply:
Ended up having to take a slightly different tack for a couple fields:
* default - `trace` a warning when there's a conflict otherwise fall back to the last set subschema's default
* deprecated - only mark the result deprecated if all subschemas are marked deprecated, otherwise null